supervisor/dbus/network/setting/generate.py

```diff
@@ -134,6 +135,16 @@ def _get_ipv6_connection_settings(ipv6setting) -> dict:
     return ipv6
 
 
+def _map_mdns_setting(mode: MulticastDnsMode | None) -> int:
+    mapping = {
+        MulticastDnsMode.OFF: MulticastDnsValue.OFF,
+        MulticastDnsMode.RESOLVE: MulticastDnsValue.RESOLVE,
+        MulticastDnsMode.ANNOUNCE: MulticastDnsValue.ANNOUNCE,
+    }
+
+    return int(mapping[mode] if mode else MulticastDnsValue.ANNOUNCE)
```

Comment:
There is actually also the mode -1 (default). Since we created the connections with the hardcoded value 2 (announce) so far, it probably hardly ever happens.

I wonder if we actually should remove the hard coded announce (2) today and use default (-1) (the default is announce (2) too, at least on HAOS since https://github.com/home-assistant/operating-system/pull/986, but it feels cleaner). Anyhow, this should be a separate PR, possible only once we start to officially deprecate old operating system versions.

But ideally we introduce the default option here already, since Supervisor might also encounter such a connection setting today. I guess this should be straight forward, essentially add `Default` with a value of `-1` to `MulticastDnsMode` and `MulticastDnsValue`.